rego/rego.go

```diff
@@ -2405,6 +2405,51 @@ func (r *Rego) partial(ctx context.Context, ectx *EvalContext) (*PartialQueries,
 		return nil, err
 	}
 
+	if r.regoVersion == ast.RegoV0 && (r.capabilities == nil || r.capabilities.ContainsFeature(ast.FeatureRegoV1Import)) {
+		// If the target rego-version in v0, and the rego.v1 import is available, then we attempt to apply it to support modules.
+
+		for i, mod := range support {
+			if mod.RegoVersion() != ast.RegoV0 {
+				continue
+			}
+
+			// We can't apply the RegoV0CompatV1 version to the support module if it contains rules or vars that
+			// conflict with future keywords.
+			applyRegoVersion := true
+
+			ast.WalkRules(mod, func(r *ast.Rule) bool {
+				name := r.Head.Name
+				if name == "" && len(r.Head.Reference) > 0 {
+					name = r.Head.Reference[0].Value.(ast.Var)
+				}
+				if ast.IsFutureKeyword(name.String()) {
+					applyRegoVersion = false
+					return true
+				}
+				return false
+			})
+
+			if applyRegoVersion {
+				ast.WalkVars(mod, func(v ast.Var) bool {
+					if ast.IsFutureKeyword(v.String()) {
+						applyRegoVersion = false
+						return true
+					}
+					return false
+				})
```

Comment:
Need to create tests for this before un-drafting this PR.

Reply:
Is it even possible to retain var names such that they conflict with v1 keywords? Or are they all rewritten? Any ideas?

Reply:
@srenatus any idea about this?

Reply:
We could do something similar for imports in the generated support module, but I believe all references would be locally rewritten to complete refs anyways ..